app/src/main/java/ar/com/wolox/android/training/ui/home/news/HomeNewsPresenter.kt

```diff
@@ -0,0 +1,49 @@
+package ar.com.wolox.android.training.ui.home.news
+
+import ar.com.wolox.android.training.model.INewsService
+import ar.com.wolox.android.training.utils.networkCallback
+import ar.com.wolox.wolmo.core.presenter.BasePresenter
+import ar.com.wolox.wolmo.networking.retrofit.RetrofitServices
+import javax.inject.Inject
+import kotlin.math.min
+
+class HomeNewsPresenter @Inject constructor(private val vRetrofitServices: RetrofitServices)
+    : BasePresenter<IHomeNewsView>() {
+
+    fun loadNews(pageIndex: Int, newsPerPage: Int){
+        val service = vRetrofitServices.getService(INewsService::class.java)
+        val call = service?.getAllNews("email")
+
+        view.progressCircleVisibilityOn()
```

Comment:
The view should decide when to show the dialog.

Reply:
Solved in next PR update.